### DISCUSSION TITLE
How to make a smooth "stop" action ?

### DISCUSSION THREAD

**Discussion by wibimaster** [matteobruni/tsparticles#4487] [Q&A] [answered]
Hi !

I'm trying to use the "fireworks" preset, but I want to programmatically start it (with `container.start()`) and stop it (with `container.stop()`).

I can't find any documentation for doing a "clean stop", I mean not making all the fireworks suddenly disappear, but stopping to do more and let those finish their animation.

Or, maybe a configuration to launch N fireworks and then terminate.

Is there a way to do this ?

Thanks !

**Comment by matteobruni** [comment, marked as answer]
You can set a life duration to the fireworks emitter so it will stop after seconds, or you can stop the emitter programmatically, but it's not documented since TypeDoc is not supporting monorepos correctly yet.

You can see a sample of the play/pause emitter here: https://codepen.io/matteobruni/pen/VwXWzJe

**Comment by wibimaster** [comment]
Perfect, thanks :)
